File: app/src/main/java/com/example/plantonic/ui/activity/splash/SplashScreenViewModel.java
package com.example.plantonic.ui.activity.splash;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.plantonic.repo.LoginRepository;

public class SplashScreenViewModel extends ViewModel {
    private final LoginRepository loginRepository = new LoginRepository();

    public LiveData<Boolean> checkIfUserExists(String userId){
        loginRepository.checkIfUserExists(userId);
        return loginRepository.userExists;
    }
}

File: app/src/main/java/com/example/plantonic/ui/activity/splash/SplashScreenViewModel.kt
package com.example.plantonic.ui.activity.splash


import androidx.lifecycle.ViewModel
import com.example.plantonic.repo.LoginRepository
import androidx.lifecycle.LiveData

class SplashScreenViewModel : ViewModel() {
    private val loginRepository = LoginRepository()


    fun checkIfUserExists(userId: String?): LiveData<Boolean?> {
        loginRepository.checkIfUserExists(userId!!)
        return loginRepository.userExists
    }

    fun getUserToken (uid : String) : LiveData<String?> {
        return loginRepository.getJwtToken(uid)
    }

}